src/vulkan-renderer/gpu_info.cpp

```diff
@@ -11,259 +11,219 @@
-
-    spdlog::debug(
-        "------------------------------------------------------------------------------------------------------------");
-    spdlog::debug("Number of queue families: {}", number_of_queue_families);
-    spdlog::debug(
-        "------------------------------------------------------------------------------------------------------------");
-
-    if (number_of_queue_families <= 0) {
-        spdlog::critical("Error: Could not find any queue families!");
-    } else {
-        // Preallocate memory for the family queues.
-        std::vector<VkQueueFamilyProperties> queue_family_properties(number_of_queue_families);
-
-        // Get information about physical device queue family properties.
-        vkGetPhysicalDeviceQueueFamilyProperties(graphics_card, &number_of_queue_families,
-                                                 queue_family_properties.data());
-
-        // Loop through all available queue families.
-        for (std::size_t i = 0; i < number_of_queue_families; i++) {
-            spdlog::debug("Queue family: {}", i);
-            spdlog::debug("--------------------------------------------------------------------------------------------"
-                          "----------------");
-            spdlog::debug("Queue Count: {}", queue_family_properties[i].queueCount);
-            spdlog::debug("Timestamp Valid Bits: {}", queue_family_properties[i].timestampValidBits);
-
-            if (queue_family_properties[i].queueFlags & VK_QUEUE_GRAPHICS_BIT) {
-                spdlog::debug("VK_QUEUE_GRAPHICS_BIT");
-            }
-            if (queue_family_properties[i].queueFlags & VK_QUEUE_COMPUTE_BIT) {
-                spdlog::debug("VK_QUEUE_COMPUTE_BIT");
-            }
-            if (queue_family_properties[i].queueFlags & VK_QUEUE_TRANSFER_BIT) {
-                spdlog::debug("VK_QUEUE_TRANSFER_BIT");
-            }
-            if (queue_family_properties[i].queueFlags & VK_QUEUE_SPARSE_BINDING_BIT) {
-                spdlog::debug("VK_QUEUE_SPARSE_BINDING_BIT");
-            }
-            if (queue_family_properties[i].queueFlags & VK_QUEUE_PROTECTED_BIT) {
-                spdlog::debug("VK_QUEUE_PROTECTED_BIT");
-            }
-
-            const std::uint32_t width = queue_family_properties[i].minImageTransferGranularity.width;
-            const std::uint32_t height = queue_family_properties[i].minImageTransferGranularity.width;
-            const std::uint32_t depth = queue_family_properties[i].minImageTransferGranularity.depth;
-
-            spdlog::debug("Min Image Timestamp Granularity: {}, {}, {}", width, height, depth);
+    std::uint32_t queue_family_count = 0;
+
+    vkGetPhysicalDeviceQueueFamilyProperties(graphics_card, &queue_family_count, nullptr);
+
+    spdlog::debug("Number of queue families: {}", queue_family_count);
+
+    if (queue_family_count == 0) {
+        throw std::runtime_error("Error: Could not find any queue families!");
```

Comment:
I don't think the info printer should throw any exceptions if there aren't any queue families, it should just say that there aren't any - this code doesn't *need* any queue families to exist, it's just printing as much info it can about them.

Reply:
Should this be applied to every method in his class then?
They all throw exceptions.

Reply:
Yeah I think so, at least if it's just iterating. If it's not safe to continue though, there should still be an exception.

Reply:
So in this case we use an else block after the if isntead of the exception?

Reply:
You don't need to put anything, you can just remove the check.

Reply:
Done.